### PR TITLE
invalidate enable() and disable() funstions

### PR DESCRIPTION
**Pull request by youtr-s**

### CODE REVIEW

File: src/js/Alert/index.ts [hunk 1]
```diff
@@ -48,6 +53,12 @@ class Alert extends Control {
     return className.join(' ');
   }
 
+  on(eventName: string, callback: (params?: any) => void) {
+    this.element.addEventListener(eventName, (e: Event)=>{
+      callback(e);
+    });
+  }
```
Comment (phongnhat19) [inline src/js/Alert/index.ts]: No need to reimplement the `on` function. We can implemented it this way:
```
on(eventName: string, callback: (params?: any) => void) {
    this._props.isDisabled = false;
    super.on(eventName, callback);
  }
```

File: src/js/Label/index.ts [hunk 2]
```diff
@@ -71,6 +77,12 @@ class Label extends Control {
     }
   }
 
+  on(eventName: string, callback: (params?: any) => void) {
+    this.element.addEventListener(eventName, (e: Event)=>{
+      callback(e);
+    });
+  }
+
```
Comment (phongnhat19) [inline src/js/Label/index.ts]: Same as above comment.

File: src/js/Alert/index.ts [hunk 3]
```diff
@@ -27,8 +27,13 @@ class Alert extends Control {
   }
 
   rerender(changedAttr?: string[]) {
-    super.rerender();
-
```
Comment (phongnhat19) [inline src/js/Alert/index.ts]: Can be implemented this way:
```
this._props.isDisabled = false;
super.rerender();

// ... rest of your function

```

File: src/js/Label/index.ts [hunk 4]
```diff
@@ -48,7 +48,13 @@ class Label extends Control {
   }
 
   rerender(changedAttr?: string[]) {
-    super.rerender();
```
Comment (phongnhat19) [inline src/js/Label/index.ts]: Same as above.

File: src/js/NotifyPopup/index.ts [hunk 5]
```diff
@@ -95,7 +95,13 @@ class NotifyPopup extends Control {
   }
 
   rerender(changedAttr?: string[]) {
-    super.rerender();
```
Comment (phongnhat19) [inline src/js/NotifyPopup/index.ts]: Same as above.

File: src/js/Spinner/index.ts [hunk 6]
```diff
@@ -28,6 +28,24 @@ class Spinner extends Control {
...
+
+  rerender(changedAttr?: string[]) {
+    if (this.element) {
+      if (!this._props.isVisible) {
+        this.element.style.display = 'none';
+      } else {
+        this.element.style.display = '';
+      }
+    }
+  }
```
Comment (phongnhat19) [inline src/js/Spinner/index.ts]: Same as above.

File: src/react/Alert/index.tsx [hunk 7]
```diff
@@ -11,6 +11,10 @@ type AlertProps = {
 }
 
 const Alert = ({text, type, isDisabled, isVisible, onClick}: AlertProps) => {
+  // isDisabled always is setted false
+  // When we update major version of ui-component, we should delete this prop
+  isDisabled = false;
```
Comment (phongnhat19) [inline src/react/Alert/index.tsx]: ESLint problem

File: src/react/Label/index.tsx [hunk 8]
```diff
@@ -12,6 +12,10 @@ type LabelProps = {
 }
 
 const Label = ({text, textColor, backgroundColor, isRequired, isDisabled, isVisible, onClick}: LabelProps) => {
+  // isDisabled always is setted false
+  // When we update major version of ui-component, we should delete this prop
+  isDisabled = false;
```
Comment (phongnhat19) [inline src/react/Label/index.tsx]: ESLint problem

File: src/react/NotifyPopup/index.tsx [hunk 9]
```diff
@@ -13,6 +13,10 @@ type NotifyPopupProps = {
 }
 
 const NotifyPopup = ({text, type, isDisabled, isVisible, onClick, onClose}: NotifyPopupProps) => {
+  // isDisabled always is setted false
+  // When we update major version of ui-component, we should delete this prop
+  isDisabled = false;
```
Comment (phongnhat19) [inline src/react/NotifyPopup/index.tsx]: ESLint problem

File: src/react/Alert/index.tsx [hunk 10]
```diff
@@ -11,6 +11,8 @@ type AlertProps = {
 }
 
 const Alert = ({text, type, isDisabled, isVisible, onClick}: AlertProps) => {
```
Comment (phongnhat19) [inline src/react/Alert/index.tsx]: `isDisabled` is not used in the component so I think it should be removed here.

File: src/react/Label/index.tsx [hunk 11]
```diff
@@ -12,10 +12,9 @@ type LabelProps = {
 }
 
 const Label = ({text, textColor, backgroundColor, isRequired, isDisabled, isVisible, onClick}: LabelProps) => {
```
Comment (phongnhat19) [inline src/react/Label/index.tsx]: `isDisabled` is not used in the component so I think it should be removed here.

File: src/react/NotifyPopup/index.tsx [hunk 12]
```diff
@@ -13,10 +13,9 @@ type NotifyPopupProps = {
 }
 
 const NotifyPopup = ({text, type, isDisabled, isVisible, onClick, onClose}: NotifyPopupProps) => {
```
Comment (phongnhat19) [inline src/react/NotifyPopup/index.tsx]: `isDisabled` is not used in the component so I think it should be removed here.

File: src/react/Alert/index.tsx (hunk 10 above)
Comment (youtr-s) [reply src/react/Alert/index.tsx]: @phongnhat19 
I think if we delete `isDisabled` immediately, people who specify this param like following in old ui-component will get compile error when update ui-component and use latest component.

`var label = new kintoneUIComponent.Label({text: 'label', isDisable: 'false'});`

So it is better to delete `isDisabled` when we update major version of ui-component.
What do you think ?

Comment (phongnhat19) [reply src/react/Alert/index.tsx]: @youtr-s 
We remove `isDisabled` from destructed props of React, so using it in JS (as your sample) won't be affected.

**Pull request merged**